### PR TITLE
Restore `spack checksum` verbosity

### PR DESCRIPTION
**Pull request by adamjstewart**
This also affects `spack create`.

### Before

```console
$ spack checksum python

==> How many would you like to checksum? (default is 1, q to abort)
```

### After

```console
$ spack checksum python
==> Found 309 versions of python:
  
  3.10.0a1   https://www.python.org/ftp/python/3.10.0/Python-3.10.0a1.tgz
  3.9.0rc2   https://www.python.org/ftp/python/3.9.0/Python-3.9.0rc2.tgz
  3.9.0rc1   https://www.python.org/ftp/python/3.9.0/Python-3.9.0rc1.tgz
  3.9.0b5    https://www.python.org/ftp/python/3.9.0/Python-3.9.0b5.tgz
  3.9.0b4    https://www.python.org/ftp/python/3.9.0/Python-3.9.0b4.tgz
  3.9.0b3    https://www.python.org/ftp/python/3.9.0/Python-3.9.0b3.tgz
  3.9.0b2    https://www.python.org/ftp/python/3.9.0/Python-3.9.0b2.tgz
  3.9.0b1    https://www.python.org/ftp/python/3.9.0/Python-3.9.0b1.tgz
  3.9.0a6    https://www.python.org/ftp/python/3.9.0/Python-3.9.0a6.tgz
  ...
  2.0.1      https://www.python.org/ftp/python/2.0.1/Python-2.0.1.tgz

==> How many would you like to checksum? (default is 1, q to abort) 
```

### CODE REVIEW

**Review by sethrj — APPROVED** [review]
I like it, but you might need someone else who's more in charge for user-facing uniformity.

**Comment by adamjstewart** [comment]
Ping @tldahlgren

**Comment by adamjstewart** [comment]
Can I get another review from a core developer?

**Review by tldahlgren — APPROVED** [review]
LGTM.

**Pull request merged**
